src/vaadin-pdf-viewer.html

```diff
@@ -111,14 +190,89 @@
             // No file given, show nothing.
             return;
           }
+          this.setFilename(src);
           this._document = pdfjsLib.getDocument(src);
           return this._document.promise.then((pdfDocument) => {
             // Document loaded, specifying document for the viewer.
             this._viewer.setDocument(pdfDocument);
+            this.setPdfTitleFromMetadata(pdfDocument);
+            this.$.toolbar.classList.add('ready');
+            this._totalPages = pdfDocument.numPages;
+            this._updatePageNumberStates();
+            this.dispatchEvent(new CustomEvent('document-loaded', {
+              detail: {
+                document: pdfDocument
+              }
+            }));
           }, function (exception) {
             console.error(exception && exception.message);
           });
         }
+
+        setFilename(src) {
+          let filename = pdfjsLib.getFilenameFromUrl(src) || src;
+          try {
+            filename = decodeURIComponent(filename);
+          } catch (e) {
+            // decodeURIComponent may throw URIError,
+            // fall back to using the unprocessed url in that case
+          }
+          this._filename = filename;
+        }
+
+        setPdfTitleFromMetadata(pdfDocument) {
+          const self = this;
+          pdfDocument.getMetadata().then(function (data) {
+            let pdfTitle;
+            const metadata = data.metadata;
+            if (metadata && metadata.has('dc:title')) {
+              const title = metadata.get('dc:title');
+              // Ghostscript sometimes returns 'Untitled', so prevent setting the
+              // title to 'Untitled'.
+              if (title !== 'Untitled') {
+                pdfTitle = title;
+              }
+            }
+
+            const info = data.info;
+            if (!pdfTitle && info && info['Title']) {
+              pdfTitle = info['Title'];
+            }
+            self._pdfTitle = pdfTitle;
+          });
+        }
+
+        _updatePageNumberStates() {
```

Comment:
Is this API intentionally protected (use __ for private)?

Same goes for the methods below

Reply:
Changed to private. Done.